File: lab4/backend/products/soa-products-rest-adapter/src/main/kotlin/com/soa/products/rest/adapter/wsdl/UnitOfMeasureTo.java

package com.soa.products.rest.adapter.wsdl;

import jakarta.xml.bind.annotation.XmlEnum;
import jakarta.xml.bind.annotation.XmlType;


/**
 * 
 * 
 * <p>Java class for unitOfMeasureTo</p>.
 * 
 * <p>The following schema fragment specifies the expected content contained within this class.</p>
 * <pre>{@code
 * <simpleType name="unitOfMeasureTo">
 *   <restriction base="{http://www.w3.org/2001/XMLSchema}string">
 *     <enumeration value="KILOGRAMS"/>
 *     <enumeration value="METERS"/>
 *     <enumeration value="MILLILITERS"/>
 *     <enumeration value="MILLIGRAMS"/>
 *   </restriction>
 * </simpleType>
 * }</pre>
 * 
 */
@XmlType(name = "unitOfMeasureTo")
@XmlEnum
public enum UnitOfMeasureTo {

    KILOGRAMS,
    METERS,
    MILLILITERS,
    MILLIGRAMS;

    public String value() {
        return name();
    }

    public static UnitOfMeasureTo fromValue(String v) {
        return valueOf(v);
    }

}

File: lab4/backend/products/soa-products/src/main/kotlin/com/soa/products/dto/UnitOfMeasureTo.kt
package com.soa.products.dto

enum class UnitOfMeasureTo(val value: String) {

    KILOGRAMS("KILOGRAMS"),

    METERS("METERS"),

    MILLILITERS("MILLILITERS"),

    MILLIGRAMS("MILLIGRAMS");

    companion object {
        fun forValue(value: String): UnitOfMeasureTo {
            return values().first { it.value == value }
        }
    }
}
